src/main/scala/com/target/data_validator/ValidatorTable.scala

```diff
@@ -210,7 +211,18 @@ case class ValidatorHiveTable(
 
   override def getDF(session: SparkSession): Try[DataFrame] = {
     logger.info(s"Opening table: $db.$table")
-    Try(session.table(s"$db.$table"))
+    useHWC match {
+      case Some(x) if x =>
+        val hive = com.hortonworks.hwc.HiveWarehouseSession.session(session).build()
+        val query = condition match {
+          case Some(c) => s"SELECT * FROM $db.$table WHERE $c"
+          case None => s"SELECT * FROM $db.$table"
+        }
+        Try(hive.executeQuery(query))
```

Comment:
See my comment above.  `hive.table(s"$db.$table")` method appears to be functionally equivalent to `hive.executeQuery(s"select * from $db.$table")`.  If pushdown worked correctly then this would be the ideal implementation.